### PR TITLE
Initial base point for version 2.0

### PR DESCRIPTION
**Pull request by ben-challis**
Key features / changes.

* PHP 7.1 minimum version
* PSR-4 autoloader
* Support for firewalls to not act under

### CODE REVIEW

File: src/Resources/config/services.yml [hunk 1]
```diff
@@ -1,12 +1,18 @@
 parameters:
-#    alpha_visitor_tracking.example.class: Alpha\VisitorTrackingBundle\Example
```
Comment (martin-georgiev) [inline src/Resources/config/services.yml]: `parameters:` can be removed, it doesn't contain any value anymore

File: src/EventListener/VisitorTrackingSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+     * @var array
+     */
+    private $firewallBlacklist;
+
+    /**
+     * @var FirewallMap
+     */
+    private $firewallMap;
+
+    public function __construct(EntityManager $em, SessionStore $sessionStore, array $firewallBlacklist, FirewallMap $firewallMap)
```
Comment (martin-georgiev) [inline src/EventListener/VisitorTrackingSubscriber.php]: We can improve the first argument's injection to be for `EntityManagerInterface`.

Comment (ben-challis) [reply src/EventListener/VisitorTrackingSubscriber.php]: We can't.

It's reliant on the `EntityManager`'s signature allowing to specify a single entity to flush. 

Dropping this behaviour is possible, uncertain if it's a good idea at present. Doctrine wants to the drop this functionality entirely however, so it's certainly something worth looking at, as we'll have to change this at some point in the future.

Comment (martin-georgiev) [reply src/EventListener/VisitorTrackingSubscriber.php]: Oh, I've missed that point. Good call!

**Pull request merged**